File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.cpp
class Solution
{
public:
  int longestConsecutive(vector<int> &nums)
  {
    unordered_set<int> s{nums.begin(), nums.end()};
    int longest = 0;
    for (auto &&num : nums)
    {
      if (!s.count(num - 1))
      {
        int length = 1;
        while (s.count(num + length))
        {
          ++length;
        }
        longest = max(length, longest);
      }
    }
    return longest;
  }
};

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.js
var longestConsecutive = function (nums) {
  const set = new Set();
  for (const num of nums) {
    set.add(num);
  }
  let longest = 0;
  for (const num of set) {
    if (!set.has(num - 1)) {
      // New sequence
      let length = 1;
      while (set.has(num + length)) {
        length++;
      }
      longest = Math.max(longest, length);
    }
  }
  return longest;
};